core/src/main/scala/org/bitcoins/core/crypto/TransactionSignatureSerializer.scala

```diff
@@ -253,8 +397,17 @@ sealed abstract class TransactionSignatureSerializer {
       errorHash
     } else {
       val serializedTxForSignature =
-        serializeForSignature(txSigComponent, hashType)
-      CryptoUtil.doubleSHA256(serializedTxForSignature)
+        serializeForSignature(txSigComponent = txSigComponent,
+                              hashType = hashType,
+                              taprootOptions = taprootOptions)
+
+      val hash = txSigComponent.sigVersion match {
+        case _: SigVersionTaproot =>
+          CryptoUtil.taggedSha256(serializedTxForSignature, "TapSighash")
```

Comment:
this should probably be given a method in `CryptoRuntime.scala` so that the tag doesn't have to be re-computed every time